Compiler warnings on type conversions

Linux CentOS 8. Compiler in question:
```
$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --disable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.3.1 20191121 (Red Hat 8.3.1-5) (GCC) 
```

Errors (warnings that are treated as errors):
```
/home/ur20980/src/PQClean/test/../crypto_kem/sntrup653/clean/crypto_core_inv3sntrup653.c:103:12: error: 'signed char' to 'int' conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
    sign = f[0];
           ^
/home/ur20980/src/PQClean/test/../crypto_kem/sntrup653/clean/crypto_core_invsntrup653.c:82:24: error: 'signed char' to 'Fq' (aka 'short') conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
        g[p - 1 - i] = in[i];
                       ^

_____________________________ test_clang_tidy[clean implementation of sntrup761] _____________________________
[gw3] linux -- Python 3.6.8 /usr/bin/python3.6

implementation = <Implementation(sntrup761, clean)>

    @pytest.mark.parametrize(
        'implementation',
        pqclean.Scheme.all_supported_implementations(),
        ids=str,
    )
    @helpers.skip_windows()
    @helpers.filtered_test
    def test_clang_tidy(implementation: pqclean.Implementation):
        helpers.ensure_available('clang-tidy')
        cfiles = implementation.cfiles()
        common_files = glob(os.path.join('..', 'common', '*.c'))
        (returncode, _) = helpers.run_subprocess(
            ['clang-tidy',
             '-quiet',
             '-header-filter=.*',
             *additional_flags,
             *cfiles,
             *common_files,
             '--',
             '-iquote', os.path.join('..', 'common'),
             '-iquote', implementation.path()],
            expected_returncode=None,
        )
    
        # Detect and gracefully avoid segfaults
        if returncode == -11:
            raise unittest.SkipTest("clang-tidy segfaulted")
    
>       assert returncode == 0, "Clang-tidy returned %d" % returncode
E       AssertionError: Clang-tidy returned 2
E       assert 2 == 0
E         +2
E         -0

test_linter.py:42: AssertionError
.  .  .  .  .
-------------------------------------------- Captured stdout call --------------------------------------------
. > clang-tidy -quiet -header-filter=.* ../crypto_kem/sntrup761/clean/crypto_core_inv3sntrup761.c ../crypto_kem/sntrup761/clean/crypto_core_invsntrup761.c ../crypto_kem/sntrup761/clean/crypto_core_mult3sntrup761.c ../crypto_kem/sntrup761/clean/crypto_core_multsntrup761.c ../crypto_kem/sntrup761/clean/crypto_core_scale3sntrup761.c ../crypto_kem/sntrup761/clean/crypto_core_weightsntrup761.c ../crypto_kem/sntrup761/clean/crypto_core_wforcesntrup761.c ../crypto_kem/sntrup761/clean/crypto_decode_761x1531.c ../crypto_kem/sntrup761/clean/crypto_decode_761x3.c ../crypto_kem/sntrup761/clean/crypto_decode_761x4591.c ../crypto_kem/sntrup761/clean/crypto_decode_761xint16.c ../crypto_kem/sntrup761/clean/crypto_decode_761xint32.c ../crypto_kem/sntrup761/clean/crypto_encode_761x1531.c ../crypto_kem/sntrup761/clean/crypto_encode_761x1531round.c ../crypto_kem/sntrup761/clean/crypto_encode_761x3.c ../crypto_kem/sntrup761/clean/crypto_encode_761x4591.c ../crypto_kem/sntrup761/clean/crypto_encode_761xfreeze3.c ../crypto_kem/sntrup761/clean/crypto_encode_761xint16.c ../crypto_kem/sntrup761/clean/crypto_encode_int16.c ../crypto_kem/sntrup761/clean/crypto_sort_int32.c ../crypto_kem/sntrup761/clean/crypto_sort_uint32.c ../crypto_kem/sntrup761/clean/crypto_stream_aes256ctr.c ../crypto_kem/sntrup761/clean/crypto_verify_1039.c ../crypto_kem/sntrup761/clean/kem.c ../common/aes.c ../common/fips202.c ../common/nistseedexpander.c ../common/randombytes.c ../common/sha2.c ../common/sp800-185.c -- -iquote ../common -iquote ../crypto_kem/sntrup761/clean
3 warnings generated.
6 warnings generated.
8 warnings generated.
10 warnings generated.
12 warnings generated.
14 warnings generated.
16 warnings generated.
18 warnings generated.
20 warnings generated.
22 warnings generated.
24 warnings generated.
26 warnings generated.
28 warnings generated.
30 warnings generated.
32 warnings generated.
34 warnings generated.
36 warnings generated.
38 warnings generated.
40 warnings generated.
42 warnings generated.
44 warnings generated.
49 warnings generated.
51 warnings generated.
65 warnings generated.
70 warnings generated.
75 warnings generated.
80 warnings generated.
108 warnings generated.
113 warnings generated.
115 warnings generated.
/home/ur20980/src/PQClean/test/../crypto_kem/sntrup761/clean/crypto_core_inv3sntrup761.c:103:12: error: 'signed char' to 'int' conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
    sign = f[0];
           ^
/home/ur20980/src/PQClean/test/../crypto_kem/sntrup761/clean/crypto_core_invsntrup761.c:82:24: error: 'signed char' to 'Fq' (aka 'short') conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
        g[p - 1 - i] = in[i];
                       ^

_____________________________ test_clang_tidy[clean implementation of sntrup857] _____________________________
[gw3] linux -- Python 3.6.8 /usr/bin/python3.6

implementation = <Implementation(sntrup857, clean)>

    @pytest.mark.parametrize(
        'implementation',
        pqclean.Scheme.all_supported_implementations(),
        ids=str,
    )
    @helpers.skip_windows()
    @helpers.filtered_test
    def test_clang_tidy(implementation: pqclean.Implementation):
        helpers.ensure_available('clang-tidy')
        cfiles = implementation.cfiles()
        common_files = glob(os.path.join('..', 'common', '*.c'))
        (returncode, _) = helpers.run_subprocess(
            ['clang-tidy',
             '-quiet',
             '-header-filter=.*',
             *additional_flags,
             *cfiles,
             *common_files,
             '--',
             '-iquote', os.path.join('..', 'common'),
             '-iquote', implementation.path()],
            expected_returncode=None,
        )
    
        # Detect and gracefully avoid segfaults
        if returncode == -11:
            raise unittest.SkipTest("clang-tidy segfaulted")
    
>       assert returncode == 0, "Clang-tidy returned %d" % returncode
E       AssertionError: Clang-tidy returned 2
E       assert 2 == 0
E         +2
E         -0

test_linter.py:42: AssertionError
-------------------------------------------- Captured stdout call --------------------------------------------
. > clang-tidy -quiet -header-filter=.* ../crypto_kem/sntrup857/clean/crypto_core_inv3sntrup857.c ../crypto_kem/sntrup857/clean/crypto_core_invsntrup857.c ../crypto_kem/sntrup857/clean/crypto_core_mult3sntrup857.c ../crypto_kem/sntrup857/clean/crypto_core_multsntrup857.c ../crypto_kem/sntrup857/clean/crypto_core_scale3sntrup857.c ../crypto_kem/sntrup857/clean/crypto_core_weightsntrup857.c ../crypto_kem/sntrup857/clean/crypto_core_wforcesntrup857.c ../crypto_kem/sntrup857/clean/crypto_decode_857x1723.c ../crypto_kem/sntrup857/clean/crypto_decode_857x3.c ../crypto_kem/sntrup857/clean/crypto_decode_857x5167.c ../crypto_kem/sntrup857/clean/crypto_decode_857xint16.c ../crypto_kem/sntrup857/clean/crypto_decode_857xint32.c ../crypto_kem/sntrup857/clean/crypto_encode_857x1723.c ../crypto_kem/sntrup857/clean/crypto_encode_857x1723round.c ../crypto_kem/sntrup857/clean/crypto_encode_857x3.c ../crypto_kem/sntrup857/clean/crypto_encode_857x5167.c ../crypto_kem/sntrup857/clean/crypto_encode_857xfreeze3.c ../crypto_kem/sntrup857/clean/crypto_encode_857xint16.c ../crypto_kem/sntrup857/clean/crypto_encode_int16.c ../crypto_kem/sntrup857/clean/crypto_sort_int32.c ../crypto_kem/sntrup857/clean/crypto_sort_uint32.c ../crypto_kem/sntrup857/clean/crypto_stream_aes256ctr.c ../crypto_kem/sntrup857/clean/crypto_verify_1184.c ../crypto_kem/sntrup857/clean/kem.c ../common/aes.c ../common/fips202.c ../common/nistseedexpander.c ../common/randombytes.c ../common/sha2.c ../common/sp800-185.c -- -iquote ../common -iquote ../crypto_kem/sntrup857/clean
3 warnings generated.
6 warnings generated.
8 warnings generated.
10 warnings generated.
12 warnings generated.
14 warnings generated.
16 warnings generated.
18 warnings generated.
20 warnings generated.
22 warnings generated.
24 warnings generated.
26 warnings generated.
28 warnings generated.
30 warnings generated.
32 warnings generated.
34 warnings generated.
36 warnings generated.
38 warnings generated.
40 warnings generated.
42 warnings generated.
44 warnings generated.
49 warnings generated.
51 warnings generated.
65 warnings generated.
70 warnings generated.
75 warnings generated.
80 warnings generated.
108 warnings generated.
113 warnings generated.
115 warnings generated.
/home/ur20980/src/PQClean/test/../crypto_kem/sntrup857/clean/crypto_core_inv3sntrup857.c:103:12: error: 'signed char' to 'int' conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
    sign = f[0];
           ^
/home/ur20980/src/PQClean/test/../crypto_kem/sntrup857/clean/crypto_core_invsntrup857.c:82:24: error: 'signed char' to 'Fq' (aka 'short') conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
        g[p - 1 - i] = in[i];
                       ^

____________________________ test_clang_tidy[clean implementation of falcon-1024] ____________________________
[gw3] linux -- Python 3.6.8 /usr/bin/python3.6

implementation = <Implementation(falcon-1024, clean)>

    @pytest.mark.parametrize(
        'implementation',
        pqclean.Scheme.all_supported_implementations(),
        ids=str,
    )
    @helpers.skip_windows()
    @helpers.filtered_test
    def test_clang_tidy(implementation: pqclean.Implementation):
        helpers.ensure_available('clang-tidy')
        cfiles = implementation.cfiles()
        common_files = glob(os.path.join('..', 'common', '*.c'))
        (returncode, _) = helpers.run_subprocess(
            ['clang-tidy',
             '-quiet',
             '-header-filter=.*',
             *additional_flags,
             *cfiles,
             *common_files,
             '--',
             '-iquote', os.path.join('..', 'common'),
             '-iquote', implementation.path()],
            expected_returncode=None,
        )
    
        # Detect and gracefully avoid segfaults
        if returncode == -11:
            raise unittest.SkipTest("clang-tidy segfaulted")
    
>       assert returncode == 0, "Clang-tidy returned %d" % returncode
E       AssertionError: Clang-tidy returned 1
E       assert 1 == 0
E         +1
E         -0

test_linter.py:42: AssertionError
-------------------------------------------- Captured stdout call --------------------------------------------
. > clang-tidy -quiet -header-filter=.* ../crypto_sign/falcon-1024/clean/codec.c ../crypto_sign/falcon-1024/clean/common.c ../crypto_sign/falcon-1024/clean/fft.c ../crypto_sign/falcon-1024/clean/fpr.c ../crypto_sign/falcon-1024/clean/keygen.c ../crypto_sign/falcon-1024/clean/rng.c ../crypto_sign/falcon-1024/clean/sign.c ../crypto_sign/falcon-1024/clean/vrfy.c ../crypto_sign/falcon-1024/clean/pqclean.c ../common/aes.c ../common/fips202.c ../common/nistseedexpander.c ../common/randombytes.c ../common/sha2.c ../common/sp800-185.c -- -iquote ../common -iquote ../crypto_sign/falcon-1024/clean
5 warnings generated.
10 warnings generated.
15 warnings generated.
20 warnings generated.
26 warnings generated.
31 warnings generated.
36 warnings generated.
41 warnings generated.
58 warnings generated.
63 warnings generated.
68 warnings generated.
73 warnings generated.
101 warnings generated.
106 warnings generated.
108 warnings generated.
/home/ur20980/src/PQClean/test/../crypto_sign/falcon-1024/clean/keygen.c:2416:13: error: 'signed char' to 'int32_t' (aka 'int') conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
        z = f[u];
            ^

____________________________ test_clang_tidy[clean implementation of falcon-512] _____________________________
[gw3] linux -- Python 3.6.8 /usr/bin/python3.6

implementation = <Implementation(falcon-512, clean)>

    @pytest.mark.parametrize(
        'implementation',
        pqclean.Scheme.all_supported_implementations(),
        ids=str,
    )
    @helpers.skip_windows()
    @helpers.filtered_test
    def test_clang_tidy(implementation: pqclean.Implementation):
        helpers.ensure_available('clang-tidy')
        cfiles = implementation.cfiles()
        common_files = glob(os.path.join('..', 'common', '*.c'))
        (returncode, _) = helpers.run_subprocess(
            ['clang-tidy',
             '-quiet',
             '-header-filter=.*',
             *additional_flags,
             *cfiles,
             *common_files,
             '--',
             '-iquote', os.path.join('..', 'common'),
             '-iquote', implementation.path()],
            expected_returncode=None,
        )
    
        # Detect and gracefully avoid segfaults
        if returncode == -11:
            raise unittest.SkipTest("clang-tidy segfaulted")
    
>       assert returncode == 0, "Clang-tidy returned %d" % returncode
E       AssertionError: Clang-tidy returned 1
E       assert 1 == 0
E         +1
E         -0

test_linter.py:42: AssertionError
-------------------------------------------- Captured stdout call --------------------------------------------
. > clang-tidy -quiet -header-filter=.* ../crypto_sign/falcon-512/clean/codec.c ../crypto_sign/falcon-512/clean/common.c ../crypto_sign/falcon-512/clean/fft.c ../crypto_sign/falcon-512/clean/fpr.c ../crypto_sign/falcon-512/clean/keygen.c ../crypto_sign/falcon-512/clean/pqclean.c ../crypto_sign/falcon-512/clean/rng.c ../crypto_sign/falcon-512/clean/sign.c ../crypto_sign/falcon-512/clean/vrfy.c ../common/aes.c ../common/fips202.c ../common/nistseedexpander.c ../common/randombytes.c ../common/sha2.c ../common/sp800-185.c -- -iquote ../common -iquote ../crypto_sign/falcon-512/clean
5 warnings generated.
10 warnings generated.
15 warnings generated.
20 warnings generated.
26 warnings generated.
43 warnings generated.
48 warnings generated.
53 warnings generated.
58 warnings generated.
63 warnings generated.
68 warnings generated.
73 warnings generated.
101 warnings generated.
106 warnings generated.
108 warnings generated.
/home/ur20980/src/PQClean/test/../crypto_sign/falcon-512/clean/keygen.c:2416:13: error: 'signed char' to 'int32_t' (aka 'int') conversion; consider casting to 'unsigned char' first. [bugprone-signed-char-misuse,-warnings-as-errors]
        z = f[u];
            ^

========================================== short test summary info ===========================================
FAILED test_linter.py::test_clang_tidy[clean implementation of sntrup653] - AssertionError: Clang-tidy retu...
FAILED test_linter.py::test_clang_tidy[clean implementation of sntrup761] - AssertionError: Clang-tidy retu...
FAILED test_linter.py::test_clang_tidy[clean implementation of sntrup857] - AssertionError: Clang-tidy retu...
FAILED test_linter.py::test_clang_tidy[clean implementation of falcon-1024] - AssertionError: Clang-tidy re...
FAILED test_linter.py::test_clang_tidy[clean implementation of falcon-512] - AssertionError: Clang-tidy ret...
========================== 5 failed, 4558 passed, 35 skipped in 7583.60s (2:06:23) ==========================
```

--- a/.clang-tidy
+++ b/.clang-tidy
@@ -1,5 +1,5 @@
 ---
-Checks:          '*,-llvm-header-guard,-hicpp-*,-readability-function-size,-google-readability-todo,-google-readability-casting,-readability-magic-numbers,-cppcoreguidelines-avoid-magic-numbers,-cppcoreguidelines-init-variables,-readability-isolate-declaration,-readability-uppercase-literal-suffix,-clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling,-clang-analyzer-valist.*,-llvmlibc-*,-bugprone-reserved-identifier,-cert-dcl37-c,-cert-dcl51-cpp'
+Checks:          '*,-llvm-header-guard,-hicpp-*,-readability-function-size,-google-readability-todo,-google-readability-casting,-readability-magic-numbers,-cppcoreguidelines-avoid-magic-numbers,-cppcoreguidelines-init-variables,-readability-isolate-declaration,-readability-uppercase-literal-suffix,-clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling,-clang-analyzer-valist.*,-llvmlibc-*,-bugprone-reserved-identifier,-cert-dcl37-c,-cert-dcl51-cpp,-bugprone-signed-char-misuse'
 WarningsAsErrors: '*'
 HeaderFilterRegex: '.*'
 AnalyzeTemporaryDtors: false

